### ISSUE TITLE
Suggestion

### ISSUE THREAD

**Issue by amirmab** [merikbest/twitter-spring-reactjs#11]
NOT ISSUE REALY, JUST A SUGGESTION
Is it not better if a service creates and exposes its clients?
for example here https://github.com/merikbest/twitter-spring-reactjs/blob/microservice/chat-service/src/main/java/com/gmail/merikbest2015/feign/TweetClient.java
you are creating tweet client, but technically that should be created in the tweet service and uploaded to artifactory as part of every release. then other services to import is use them

**Comment by Akinyemi-samuel** [comment]
Hi there,
JUST A SUGGESTION
By creating the tweet client inside the tweet service like you suggested could introduce a form of tight coupling or high coupling of dependency between services.
This would mean that changes in the Tweet service could potentially impact the Chat service, and vice versa.
In the context of microservices and software architecture, it's generally preferred to achieve loose coupling or low coupling.

** this is a suggestion **

**Comment by amirmab** [comment]
You are not wrong but some times it is needed, like if you use service X most likely will try using their SDK too right?  
Also service change should be backward compatible, it is not always possible but better to be.
// I said these based on my previous involvements with real-lworld microservices.
